File: src/rocketmqtt-admin-web/src/api/topic.ts
import { BasePageQueryRequestModel, PageQueryResponseModel } from '@/types/common'
import axios from 'axios'

export interface PageRequest extends BasePageQueryRequestModel {
    topicName: string;
}

export interface PageResult {
    topicName: string;
    node: string;
}

export function getPage(data: BasePageQueryRequestModel) {
    return axios.post<PageQueryResponseModel<PageResult>>('/api/Topic/PageList', data);
}

File: src/Api/RocketMqtt.Domain/Domain/Topic.cs
namespace RocketMqtt.Domain.Domain;

/// <summary>
/// 主题
/// </summary>
public class Topic : EntityBase
{
    /// <summary>
    /// 主题名称
    /// </summary>
    public string TopicName { get; set; }

    /// <summary>
    /// 节点
    /// </summary>
    public string Node { get; set; }

    public Topic()
    {
    }

    public Topic(string topicName, string node) : this()
    {
        TopicName = topicName;
        Node = node;
    }
}